functions/executor/call.py

```diff
@@ -104,33 +111,183 @@ def exec_dag_function(pusher_cache, kvs, triggers, function, schedule):
         l = LWWPairLattice(generate_timestamp(0), serialize_val(result))
         kvs.put(schedule.response_id, l)
 
+def _exec_func_normal(kvs, func, args):
+    func_args = ()
 
-def _process_args(arg_list):
-    return list(map(lambda v: get_serializer(v.type).load(v.body), arg_list))
+    to_resolve = []
 
+    # resolve any references to KVS objects
+    key_index_map = {}
+    for i, arg in enumerate(args):
+        if isinstance(arg, FluentReference):
+            to_resolve.append(arg)
+            key_index_map[arg.key] = i
+        func_args += (arg,)
+
+    if len(to_resolve) > 0: 
+        kv_pairs = _resolve_ref_normal(to_resolve, kvs)
+
+        for key in kv_pairs:
+            func_args[key_index_map[key]] = kv_pairs[key]
+
+    # execute the function
+    return  func(*func_args)
```

Comment:
This will not work. Your new version of `_resolve_ref_normal` doesn't deserialize values. See the [existing version of the code](https://github.com/fluent-project/fluent/blob/master/functions/executor/call.py#L126-L135).

Reply:
fixed